lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -612,21 +612,41 @@ class TraceProcessor {
     // Find the inspected frame
     const mainFrameIds = this.findMainFrameIds(keyEvents);
 
-    const frames = keyEvents
+    /** @type {Map<string, {id: string, url: string, parent?: string}>} */
+    const framesById = new Map();
+
+    // Begin collection of frame tree information with TracingStartedInBrowser,
+    // which should be present even without navigations.
+    const tracingStartedFrames = keyEvents
+        .find(e => e.name === 'TracingStartedInBrowser')?.args?.data?.frames;
+    if (tracingStartedFrames) {
+      for (const frame of tracingStartedFrames) {
+        framesById.set(frame.frame, {
+          id: frame.frame,
+          url: frame.url,
+          parent: frame.parent,
+        });
+      }
+    }
+
+    // Update known frames if FrameCommittedInBrowser events come in, typically
+    // with updated `url` (as well as pid, etc). Some traces (like timespans) may
+    // not have any committed frames.
```

Comment:
somewhat debatable if we really need to "update" the known frames if there are `FrameCommittedInBrowser` events, because only the `url` changes if there are entries for a frame in both `TracingStartedInBrowser` and a `FrameCommittedInBrowser` event, and as far as I can tell we never use those `urls` anywhere :)

However, it seems good to do because it makes sense conceptually, not updating doesn't really make this code any simpler, this is like a few dozen events at most, so 🤷